### PR TITLE
feat(bundleTarget): bundle to flat root

### PR DESCRIPTION
**Pull request by marbemac**

### CODE REVIEW

File: src/__tests__/bundle.spec.ts [hunk 1]
```diff
@@ -1,8 +1,10 @@
 import { cloneDeep } from 'lodash';
 
-import { BUNDLE_ROOT, bundleTarget } from '../bundle';
+import { BUNDLE_ROOT as BUNDLE_ROOT_POINTER, bundleTarget } from '../bundle';
 import { safeStringify } from '../safeStringify';
 
+const BUNDLE_ROOT = BUNDLE_ROOT_POINTER.slice(2);
```
Comment (P0lip) [inline src/__tests__/bundle.spec.ts:6]: Was easier than updating all tests :sweat_smile:

File: src/bundle.ts [hunk 2]
```diff
@@ -7,88 +7,141 @@ import { pathToPointer } from './pathToPointer';
...
+  const _bundle = (
+    path: string,
+    stack: Dictionary<boolean>,
+    cur?: unknown,
+    bundledRefInventory: any = {},
+    bundledObj: any = {},
+    errorsObj: any = {},
+  ) => {
+    const $refTarget = pointerToPath(path);
+    // const $bundleKey = $refTarget[$refTarget.length - 1];
```
Comment (P0lip) [inline src/bundle.ts:47]: We can probably remove it? or does it provide you any value Marc?

File: src/bundle.ts [hunk 3]
```diff
@@ -7,88 +7,141 @@ import { pathToPointer } from './pathToPointer';
...
-export const ERRORS_ROOT = '__errors__';
-
-export const bundleTarget = <T = unknown>({ document, path }: { document: T; path: string }, cur?: unknown) =>
-  _bundle(cloneDeep(document), path, { [path]: true }, cur);
-
-const _bundle = (
-  document: unknown,
-  path: string,
-  stack: Dictionary<boolean>,
+export const BUNDLE_ROOT = '#/__bundled__';
```
Comment (marbemac) [inline src/bundle.ts:10]: Hmm this is a breaking change though - any reason in particular to do it?

File: src/bundle.ts (hunk 2 above)
Comment (marbemac) [reply src/bundle.ts:47]: yeah can remove

File: src/bundle.ts (hunk 3 above)
Comment (P0lip) [reply src/bundle.ts:10]: Support nested paths.

Comment (P0lip) [reply src/bundle.ts:10]: We _could_ convert it within `bundleTarget` fn, though.
In any case, it felt weird that we accept a pointer in one case, and not the other.

File: src/bundle.ts (hunk 2 above)
Comment (P0lip) [reply src/bundle.ts:47]:
```suggestion
```

**Pull request merged**